### ISSUE TITLE
09:23:31 [ERROR] invalid literal for int() with base 10: '$PORT'

### ISSUE THREAD

**Issue by 4n70w4** [mrsuh/php-load-test#1]
`cd tests && docker run -v $(pwd):/var/loadtest -v $HOME/.ssh:/root/.ssh --net host -it direvius/yandex-tank`

```
09:23:31 [INFO] /etc/yandex-tank is not accessible to get configs list
/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Phantom/reader.py:14: FutureWarning: The pandas.parser module is deprecated and will be removed in a future version. Please import from pandas.io.parser instead
  from pandas.parser import CParserError
09:23:31 [INFO] Loading plugins...
09:23:31 [INFO] Performing test
09:23:31 [INFO] Configuring plugins...
09:23:31 [INFO] Exception: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/consoleworker.py", line 421, in perform_test
    self.core.plugins_configure()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 232, in plugins_configure
    plugin.configure()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Phantom/plugin.py", line 70, in configure
    self.phantom_config = self.phantom.config_file
    self._phantom.read_config()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Phantom/utils.py", line 95, in read_config
    stream.read_config()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Phantom/utils.py", line 275, in read_config
    self.address, do_test_connect, explicit_port)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/common/util.py", line 512, in resolve
    port = int(parts[1])
ValueError: invalid literal for int() with base 10: '$PORT'

09:23:31 [ERROR] invalid literal for int() with base 10: '$PORT'
09:23:31 [INFO] Trying to shutdown gracefully...
09:23:31 [INFO] Finishing test...
09:23:31 [INFO] Stopping load generator and aggregator
09:23:32 [INFO] Finishing monitoring
09:23:32 [ERROR] Failed finishing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7fd6536966d0>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 295, in plugins_end_test
    retcode = plugin.end_test(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 305, in end_test
    self.__save_conf()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 467, in __save_conf
    self.lp_job.send_config_snapshot(self.core.cfg_snapshot)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 563, in lp_job
    self._lp_job = self.__get_lp_job()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 575, in __get_lp_job
    api_client = self.__get_api_client()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 540, in __get_api_client
    self._api_token = self.read_token(self.get_option("token_file"))
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 629, in read_token
    "token length is %d bytes", filename, len(str(data)))
  File "/usr/local/lib/python2.7/dist-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
09:23:32 [INFO] Post-processing test...
09:23:32 [ERROR] Failed post-processing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7fd6536966d0>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 313, in plugins_post_process
    retcode = plugin.post_process(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 316, in post_process
    self.events.join()
  File "/usr/lib/python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
RuntimeError: cannot join thread before it is started
09:23:32 [INFO] Artifacts dir: /var/loadtest/logs/2019-03-04_09-23-31.yz7n8l
09:23:32 [INFO] Done graceful shutdown
09:23:32 [INFO] Close allocated resources...
09:23:32 [INFO] Done performing test with code 1
```

**Comment by 4n70w4** [comment]
Okay, i manuaally replace   `address: $SERVER:$PORT` in load.yaml

But next error:

```
09:27:43 [INFO] Installing monitoring agent at $USERNAME@$server...
09:27:43 [INFO] Creating temp dir on $server
09:27:43 [INFO] Execute on $server: /usr/bin/env python2 -c "import tempfile; print tempfile.mkdtemp();"
09:27:43 [ERROR] Failed to install monitoring agent to $server
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Telegraf/client.py", line 209, in install
    out, errors, err_code = self.ssh.execute(cmd)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/common/util.py", line 70, in execute
    with self.connect() as client:
  File "/usr/local/lib/python2.7/dist-packages/yandextank/common/util.py", line 40, in connect
    timeout=self.timeout, )
  File "/usr/local/lib/python2.7/dist-packages/paramiko/client.py", line 329, in connect
    to_try = list(self._families_and_addresses(hostname, port))
  File "/usr/local/lib/python2.7/dist-packages/paramiko/client.py", line 200, in _families_and_addresses
    hostname, port, socket.AF_UNSPEC, socket.SOCK_STREAM)
gaierror: [Errno -2] Name or service not known
09:27:59 [INFO] Checking tank resources...
09:27:59 [ERROR] There were errors:
df: no file systems processed

09:27:59 [INFO] Exception: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/consoleworker.py", line 422, in perform_test
    self.core.plugins_prepare_test()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 240, in plugins_prepare_test
    plugin.prepare_test()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 229, in prepare_test
    lp_job = self.lp_job
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 563, in lp_job
    self._lp_job = self.__get_lp_job()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 575, in __get_lp_job
    api_client = self.__get_api_client()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 540, in __get_api_client
    self._api_token = self.read_token(self.get_option("token_file"))
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 629, in read_token
    "token length is %d bytes", filename, len(str(data)))
  File "/usr/local/lib/python2.7/dist-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)

09:27:59 [ERROR] 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
09:27:59 [INFO] Trying to shutdown gracefully...
09:27:59 [INFO] Finishing test...
09:27:59 [INFO] Stopping load generator and aggregator
09:27:59 [INFO] Finishing monitoring
09:27:59 [ERROR] Failed finishing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7f1b4d5c46d0>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 295, in plugins_end_test
    retcode = plugin.end_test(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 305, in end_test
    self.__save_conf()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 467, in __save_conf
    self.lp_job.send_config_snapshot(self.core.cfg_snapshot)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 563, in lp_job
    self._lp_job = self.__get_lp_job()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 575, in __get_lp_job
    api_client = self.__get_api_client()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 540, in __get_api_client
    self._api_token = self.read_token(self.get_option("token_file"))
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 629, in read_token
    "token length is %d bytes", filename, len(str(data)))
  File "/usr/local/lib/python2.7/dist-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
09:27:59 [INFO] Post-processing test...
09:27:59 [ERROR] Failed post-processing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7f1b4d5c46d0>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 313, in plugins_post_process
    retcode = plugin.post_process(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 316, in post_process
    self.events.join()
  File "/usr/lib/python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
RuntimeError: cannot join thread before it is started
09:27:59 [INFO] Artifacts dir: /var/loadtest/logs/2019-03-04_09-27-38.fJ8XJZ
09:27:59 [INFO] Done graceful shutdown
09:27:59 [INFO] Close allocated resources...
09:27:59 [INFO] Done performing test with code 1
```

How to fill monitor.xml instead  `<Host address="$SERVER" interval="1" username="$USERNAME">
` ?

**Comment by 4n70w4** [comment]
Okay, I set in load.yaml section:
```
telegraf:
  enabled: false
```

But next error:

```
09:47:05 [INFO] /etc/yandex-tank is not accessible to get configs list
/usr/local/lib/python2.7/dist-packages/yandextank/plugins/Phantom/reader.py:14: FutureWarning: The pandas.parser module is deprecated and will be removed in a future version. Please import from pandas.io.parser instead
  from pandas.parser import CParserError
09:47:05 [INFO] Loading plugins...
09:47:05 [INFO] Performing test
09:47:05 [INFO] Configuring plugins...
09:47:05 [INFO] Testing connection to resolved address 192.168.1.39 and port 8000
09:47:05 [INFO] Resolved 192.168.1.39 into 192.168.1.39:8000
09:47:05 [INFO] Configuring StepperWrapper...
09:47:05 [INFO] Using cached stpd-file: ./logs/_572ad719d18582c0a7bc87f25f9e9ba3.stpd
09:47:05 [INFO] rps_schedule is set. Overriding cached instances param from config: 1000
09:47:05 [INFO] Monitoring plugin is not enabled313 Krps
09:47:05 [INFO] Preparing test...
09:47:06 [INFO] Checking tank resources...
09:47:06 [ERROR] There were errors:
df: no file systems processed

09:47:06 [INFO] Exception: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/consoleworker.py", line 422, in perform_test
    self.core.plugins_prepare_test()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 240, in plugins_prepare_test
    plugin.prepare_test()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 229, in prepare_test
    lp_job = self.lp_job
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 563, in lp_job
    self._lp_job = self.__get_lp_job()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 575, in __get_lp_job
    api_client = self.__get_api_client()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 540, in __get_api_client
    self._api_token = self.read_token(self.get_option("token_file"))
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 629, in read_token
    "token length is %d bytes", filename, len(str(data)))
  File "/usr/local/lib/python2.7/dist-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)

09:47:06 [ERROR] 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
09:47:06 [INFO] Trying to shutdown gracefully...
09:47:06 [INFO] Finishing test...
09:47:06 [INFO] Stopping load generator and aggregator
09:47:06 [INFO] Can't get monitoring config
09:47:06 [ERROR] Failed finishing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7ffa78b65790>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 295, in plugins_end_test
    retcode = plugin.end_test(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 305, in end_test
    self.__save_conf()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 467, in __save_conf
    self.lp_job.send_config_snapshot(self.core.cfg_snapshot)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 563, in lp_job
    self._lp_job = self.__get_lp_job()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 575, in __get_lp_job
    api_client = self.__get_api_client()
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 540, in __get_api_client
    self._api_token = self.read_token(self.get_option("token_file"))
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 629, in read_token
    "token length is %d bytes", filename, len(str(data)))
  File "/usr/local/lib/python2.7/dist-packages/future/types/newstr.py", line 102, in __new__
    return super(newstr, cls).__new__(cls, value)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
09:47:06 [INFO] Post-processing test...
09:47:06 [ERROR] Failed post-processing plugin <yandextank.plugins.DataUploader.plugin.Plugin object at 0x7ffa78b65790>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/yandextank/core/tankcore.py", line 313, in plugins_post_process
    retcode = plugin.post_process(retcode)
  File "/usr/local/lib/python2.7/dist-packages/yandextank/plugins/DataUploader/plugin.py", line 316, in post_process
    self.events.join()
  File "/usr/lib/python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
RuntimeError: cannot join thread before it is started
09:47:06 [INFO] Artifacts dir: /var/loadtest/logs/2019-03-04_09-47-05.283TD5
09:47:06 [INFO] Done graceful shutdown
09:47:06 [INFO] Close allocated resources...
09:47:06 [INFO] Done performing test with code 1
```

**Comment by mrsuh** [comment]
To tests you need to:
+ copy tests/load.yaml from tests/load.yaml.example
and replace $SERVER and $PORT variables in tests/load.yaml

+ copy tests/monitoring.xml from tests/monitoring.xml.example
and replace $SERVER and $USERNAME variables in tests/monitoring.xml

+ get token from https://overload.yandex.net and put it to file tests/overload_token.txt

+ put ssh public  key to target server.

+ run yandex-tank:
`cd tests && docker run -v $(pwd):/var/loadtest -v $HOME/.ssh:/root/.ssh --net host -it direvius/yandex-tank`

If you have errors with Yandex tank - you need to describe the problem there https://github.com/yandex/yandex-tank
